<launch>
	<param name="use_sim_time" value="true" />

	<!-- Nodelet manager -->
	<node pkg="nodelet" type="nodelet" args="manager" name="Manager" output="screen" />

	<!-- Convert depth image to point cloud -->
	<node pkg="nodelet" type="nodelet" name="depth_image_proc" args="load depth_image_proc/point_cloud_xyz Manager" output="screen">
		<remap from="camera_info" to="camera/depth_registered/camera_info"/>
		<remap from="image_rect" to="camera/depth_registered/image_raw"/>
		<remap from="points" to="filtered_points"/>
	</node>

	<!-- Mapper -->
	<rosparam file="$(find slam3d_ros)/param/one_loop_mapper2d.yaml"/>
	<param name="icp_config_file" value="$(find slam3d_ros)/param/one_loop_icp.yaml" />

	<node name="Mapper" pkg="slam3d_ros" type="mapper2d_node" >
		<remap from="pointcloud" to="filtered_points" />
		<remap from="scan" to="scan" />
	</node>

	<!-- RVIZ to view the visualization -->
	<node name="RVIZ" pkg="rviz" type="rviz" args=" -d $(find slam3d_ros)/launch/one_loop_2d.rviz" />

</launch>

